affine.ts

```diff
@@ -75,19 +56,14 @@ namespace microcode {
             return aff
         }
 
-        public recalc(force = false) {
-            if (this.dirty || force) {
-                this.dirty_ = false
-                if (this.parent) {
-                    Vec2.TranslateToRef(
-                        this.localPos_,
-                        this.parent.worldPos,
-                        this.worldPos_
-                    )
-                } else {
-                    this.worldPos_.copyFrom(this.localPos)
-                }
+        private computeWorldPos(): Vec2 {
+            const pos = this.localPos_
+            let parent = this.parent_
+            while (parent) {
+                Vec2.TranslateToRef(pos, parent.localPos, pos)
```

Comment:
Oh yes. I didn't catch that in local testing. I assumed that because it was recompiling, it was also hot loading.

After pulling your change I noticed another problem that relates to the removal of the `tag` field. It was there to handle a special case and now that it is gone, navigating up to the toolbar from the rule set is messed up. I should be able to look into this further later in the week.

Reply:
thanks. we should be able to handle that special case in the editor.

Reply:
Ok, I made a fix for that special case. If you see any more regressions please let me know. Otherwise, I believe this is ready to be merged.